docs/src/developer/gpu_computations.md

```diff
@@ -0,0 +1,58 @@
+# GPU computations
+
+GPU computations in DFTK is very much a work in progress. The goal is to build on
+Julia's multiple dispatch to have the same code base for CPU and GPU. We hope we
+will be able to reach decent performances without having to write any kernels at all,
+only by relying on the high level functionalities implemented in the GPU packages.
+
+To go even further with this idea of unified code, we would also like to be able to
+support any type of GPU architecture: unlike other packages, we do not want to
+hard-code the use of a specific architecture, say a NVIDIA CUDA GPU. In order to do
+this, DFTK uses [GPUArrays](https://github.com/JuliaGPU/GPUArrays.jl), which is the
+counterpart of `AbstractArray` but for GPU arrays.
+
+Currently the three main GPU architectures are [CUDA](https://github.com/JuliaGPU/CUDA.jl),
+[ROCm](https://github.com/JuliaGPU/AMDGPU.jl) and [OneAPI](https://github.com/JuliaGPU/oneAPI.jl).
+
+## Current implementation
+
+For now, GPU computations are done by specializing the `architecture` keyword argument
+when creating the basis. `architecture` should be an initialized instance of
+the (non-exported) `CPU` and `GPU` structures. `CPU` does not require any argument,
+but `GPU` requires the type of array which will be used for GPU computations.
+
+```@example gpu_computations
+PlaneWaveBasis(model; Ecut, kgrid, architecture = DFTK.CPU())
+PlaneWaveBasis(model; Ecut, kgrid, architecture = DFTK.GPU(CuArray))
+```
+Those `CPU` and `GPU` structures we inspired by the structures of same name in
+[Oceananigans](https://github.com/CliMA/Oceananigans.jl). **However, it is very**
```

Comment:
Well sort of. I would not put Oceananigans here as we got inspired from a couple of directions.